lib/WebpackConfig.js

```diff
@@ -298,12 +298,8 @@ class WebpackConfig {
         Object.assign(this.aliases, aliases);
     }
 
-    addExternals(externals = {}) {
-        if (typeof externals !== 'object') {
-            throw new Error('Argument 1 to addExternals() must be an object.');
-        }
-
-        Object.assign(this.externals, externals);
+    addExternals(...externals) {
```

Comment:
Nitpicking but I'm not sure that rest parameters are needed there since this is not part of the exposed API.
It adds a bit of unneeded complexity to the call in `index.js` (you have to use `apply` instead of just doing `webpackConfig.addExternals(externals)`).

Reply:
Do you mean Babel? If that's the case it's not relevant here since our code don't go through Babel.

Having `...externals`may be a bit easier to understand (and Node probably optimize it... and even if that's not the case it won't be a big overhead anyway :)) but I still don't really see the benefits of using rest parameters here instead of simply passing the array directly.

Reply:
The reason for using rest parameters was to simplify the API and the function itself. By using rest parameters or the `arguments` array, there's no need to check if the parameter is an array in order to perform the concatenation. Users can still pass an array if they want, it will just add the array to the externals list instead of appending each element. Both mechanisms work.

I meant Babel, but seeing as how it is not used, it's irrelevant. Nonetheless I would hope Node internally just converts the ES6 syntax to an `apply` call. I will change the code to the ES6 syntax to be consistent with the new API.

Reply:
Hmm, I agree with @Lyrkan. Look at the "mixed" syntax: https://webpack.js.org/configuration/externals/#combining-syntaxes

Basically, when you call `addExternal()`, there should only be 3 valid things that you can pass to it:

```js
Encore
     // an object, though the value of each entry can be a string, array or object - but we don't care
     .addExternal({jquery: 'jQuery'})

    // a function
    .addExternal(function(context, request, callback) { /*... */ })

    // regex
    .addExternal(/^(jquery|\$)$/i)
```

The current implementation allows you to pass these are multiple arguments for multiple externals. That's nice - but I'd rather just accept 1 argument and check that it's one of these 3 types.

@deAtog Could you make that change? If not - just let us know. Sorry for the delay :)